rpi-update

```diff
@@ -416,9 +427,22 @@ function do_backup {
 }
 
 function do_update {
-	if [ -f ${FW_PATH}/kernel7l.img ] || [ -f ${FW_PATH}/kernel8.img ]; then
-		WANT_PI4=1
+	if [ -f ${FW_PATH}/kernel.img ] || [ -f ${FW_PATH}/kernel7.img ]; then
+		WANT_32BIT=${WANT_32BIT:-1}
+	fi
+	if [ -f ${FW_PATH}/kernel7l.img ]; then
+		WANT_32BIT=${WANT_32BIT:-1}
+	fi
+	if [ -f ${FW_PATH}/kernel8.img ]; then
+		WANT_64BIT=${WANT_64BIT:-1}
+	fi
+	if [ -f ${FW_PATH}/start4.elf ]; then
+		WANT_PI4=${WANT_PI4:-1}
 	fi
+	WANT_32BIT=${WANT_32BIT:-0}
+	WANT_64BIT=${WANT_64BIT:-0}
+	WANT_PI4=${WANT_PI4:-0}
+	echo "WANT_32BIT:${WANT_32BIT} WANT_64BIT:${WANT_64BIT} WANT_PI4:${WANT_PI4}"
```

Comment:
Is this debug output still wanted?

Reply:
I think it is worth leaving in at least for a while.
It's possible the change in behaviour will be unwanted in some (currently unthought of) situation and this will help diagnosis.